### PR TITLE
Fix BigQuery Storage default params

### LINKED ISSUE #2690: @BigQueryType.toTable annotated class can not be used as input for `typedBigQueryStorage`
```scala
  @BigQueryType.toTable
  case class Data(name: String)

  sc.typedBigQueryStorage[Data](Table.Spec(???))
```

fails with exception:
```
INFO: Exception in thread "main" java.util.NoSuchElementException: None.get
INFO: 	at scala.None$.get(Option.scala:529)
INFO: 	at scala.None$.get(Option.scala:527)
INFO: 	at com.spotify.scio.bigquery.syntax.ScioContextOps$.typedBigQueryStorage$extension1(ScioContextSyntax.scala:238)
```

Expected behaviour:
I can define case class with fields (not generated) and use it for reading BigQuery data.

Scio version 0.8.1.

### PR DESCRIPTION
**Pull request by regadas**
Also fixes #2690

### CODE REVIEW

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/syntax/ScioContextSyntax.scala [hunk 1]
```diff
@@ -247,8 +248,8 @@ final class ScioContextOps(private val self: ScioContext) extends AnyVal {
     val table = Table.Spec(bqt.table.get)
     self.read(BigQueryTyped.Storage[T](table))(
       BigQueryTyped.Storage.ReadParam(
-        bqt.selectedFields.get,
-        rowRestriction
+        bqt.selectedFields.getOrElse(BigQueryStorage.ReadParam.DefaultSelectFields),
+        Some(rowRestriction)
```
Comment (jto) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/syntax/ScioContextSyntax.scala]: I'd rather use `Option.apply`

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/syntax/ScioContextSyntax.scala [hunk 2]
```diff
@@ -259,8 +260,8 @@ final class ScioContextOps(private val self: ScioContext) extends AnyVal {
   ): SCollection[T] =
     self.read(BigQueryTyped.Storage[T](table))(
       BigQueryTyped.Storage.ReadParam(
-        BigQueryType[T].selectedFields.get,
-        rowRestriction
+        BigQueryType[T].selectedFields.getOrElse(BigQueryStorage.ReadParam.DefaultSelectFields),
+        Some(rowRestriction)
```
Comment (jto) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/syntax/ScioContextSyntax.scala]: Same

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/syntax/ScioContextSyntax.scala (hunk 1 above)
Comment (regadas) [reply scio-bigquery/src/main/scala/com/spotify/scio/bigquery/syntax/ScioContextSyntax.scala]: never hurts being on the safe side

Comment (jto) [reply scio-bigquery/src/main/scala/com/spotify/scio/bigquery/syntax/ScioContextSyntax.scala]: wdym ? `Option.apply` is the safer one

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/taps.scala [hunk 3]
```diff
@@ -133,8 +132,8 @@ final case class BigQueryTaps(self: Taps) {
       () =>
         BigQueryStorage(Table.Ref(table)).tap(
           BigQueryStorage.ReadParam(
-            readOptions.getSelectedFieldsList().asScala.toList,
-            readOptions.getRowRestriction()
+            readOptions.getSelectedFieldsList.asScala.toList,
+            Some(readOptions.getRowRestriction)
```
Comment (jto) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/taps.scala]: `Option` :)

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/taps.scala [hunk 4]
```diff
@@ -151,8 +150,8 @@ final case class BigQueryTaps(self: Taps) {
         BigQueryStorage(Table.Ref(table))
           .tap(
             BigQueryStorage.ReadParam(
-              readOptions.getSelectedFieldsList().asScala.toList,
-              readOptions.getRowRestriction()
+              readOptions.getSelectedFieldsList.asScala.toList,
+              Some(readOptions.getRowRestriction)
```
Comment (jto) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/taps.scala]: Option :)

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/taps.scala (hunk 3 above)
Comment (regadas) [reply scio-bigquery/src/main/scala/com/spotify/scio/bigquery/taps.scala]: this one is actually ok coz getRowRestriction returns empty String if not set

Comment (regadas) [reply scio-bigquery/src/main/scala/com/spotify/scio/bigquery/taps.scala]: changed it coz on a second thought .. i don't wanna rely on third party behavior when i can play safe

**Pull request merged**